src/modules/user/Register.jsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from "react"; // Removed useContext
+import { Container } from "../../shared-component/Container/index";
+import { Input } from "../../shared-component/Input/index";
+import { useUserDispatch } from "./UserContext";
+import { useNavigate } from "react-router-dom";
+import { Button } from "../../shared-component/Button/index";
+
+const validateEmail = (email) => {
```

Comment:
You define this function twice in 2 different places. Instead, you need to place it in a shared location (e.g. src/utils/form-utils.ts) and use it from there.